### DISCUSSION TITLE
Action on realtime disconnect or after a set period of time.

### DISCUSSION THREAD

**Discussion by vin-ce** [supabase/supabase#730] [Questions] [answered]
Hello,
I was wondering if it was possible to, for example, delete a row on user socket disconnect or after 60 minutes of inactivity. 
Thank you!

**Comment by kiwicopple** [comment, marked as answer]
👋 just trying to guess the intent of you question: Are you trying to determine when somebody is "online" (for things like chat apps)?

If so, this is called "Presence". It's something we will offer in the future, but haven't built yet

To answer your question more specifically - this should be possible to do like this:

```js
    const mySubscription = supabase
    .from('*')
    .on('*', (payload) => {
      console.log('Change received!', payload)
    })
    .subscribe()

    mySubscription.onClose(() => {
      // Do something
    })
```

The reason why this isn't a great idea for "Presence" use-cases is because there could be a failure to "delete the row on disconnect". Ideally you want the server to detect "is connected", and then if a client isn't connected after XX seconds it will mark the client as "disconnected"

**Reply by vin-ce** [reply]
Yes, exactly what you said (I'm trying to keep track of and display mouse cursors). 
Thank you for the code sample! I'll implement it and add some other workarounds while I wait for the Presence implementation :)
